bothan-coingecko/src/api/error.rs

```diff
@@ -1,53 +1,23 @@
 #[derive(Clone, Debug, PartialEq, thiserror::Error)]
-/// Errors that can occur while building the `CoinGeckoRestAPI`.
-pub enum BuilderError {
-    /// An error occurred with the `reqwest` client.
-    #[error("reqwest error: {0}")]
-    Reqwest(String),
-
-    /// An invalid header value was provided.
-    #[error("invalid header value")]
+pub enum BuildError {
+    #[error("invalid header value: {0}")]
     InvalidHeaderValue(String),
 
-    /// The URL provided is invalid.
     #[error("invalid url")]
     InvalidURL(#[from] url::ParseError),
-}
-
-impl From<reqwest::header::InvalidHeaderValue> for BuilderError {
-    fn from(e: reqwest::header::InvalidHeaderValue) -> Self {
-        BuilderError::InvalidHeaderValue(e.to_string())
-    }
-}
 
-impl From<reqwest::Error> for BuilderError {
-    fn from(e: reqwest::Error) -> Self {
-        BuilderError::Reqwest(e.to_string())
-    }
+    #[error("build failed with error: {0}")]
+    BuildFailed(String),
 }
 
 #[derive(Clone, Debug, PartialEq, thiserror::Error)]
-/// Errors that can occur while interacting with the REST API.
-pub enum RestAPIError {
-    /// An HTTP error occurred.
-    #[error("http error: {0}")]
-    Http(reqwest::StatusCode),
+pub enum SendError {
+    #[error("failed to send request with error: {0}")]
+    FailedRequest(String),
 
-    /// An invalid ID was provided.
-    #[error("invalid id")]
-    InvalidID,
-
-    /// Failed to parse the response.
-    #[error("failed to parse")]
-    Parse,
-
-    /// An error occurred with the `reqwest` client.
-    #[error("reqwest error: {0}")]
-    Reqwest(String),
-}
+    #[error("received non-2xx http status: {0}")]
+    UnsuccessfulResponse(reqwest::StatusCode),
 
-impl From<reqwest::Error> for RestAPIError {
-    fn from(e: reqwest::Error) -> Self {
-        RestAPIError::Reqwest(e.to_string())
-    }
+    #[error("failed to parse with error: {0}")]
+    ParseFailed(String),
```

Comment:
You mean parse the response? Maybe add more detail.

Reply:
this error is unique to function next(), being more verbose here is not necessary in my opinion